### PR TITLE
chore: back port CI changes

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7135)

### CODE REVIEW

**Comment by coryan** [comment]
It seems like there are two unrelated changes in this PR. One is a change of the CI scripts to backport recent changes to `v1.30.x`, another is the change described here to change the CMake targets.  I think we should split them in two PRs, thoughts?

**Review by scotthart — COMMENTED** [review]
The only reason I am back porting the scripts is so that CI runs successfully with my cmake changes.

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7135#-:-MggYLX8CL-8TVKJ6zAZ:b-qfc37i)* status: 0 of 17 files reviewed, all discussions resolved

File: ci/kokoro/macos/builds/bazel.sh [hunk 1]
```diff
@@ -84,8 +83,7 @@ bazelisk build "${bazel_args[@]}" ...
 
 should_run_integration_tests() {
   if [[ -r "${GOOGLE_APPLICATION_CREDENTIALS}" && -r \
-    "${TEST_KEY_FILE_JSON}" && -r \
-    "${TEST_KEY_FILE_P12}" ]]; then
```
Comment (devjgm) [inline ci/kokoro/macos/builds/bazel.sh:88]: fyi, the p12 key files do not exist anymore, so this test will be false and integration tests will not be run.

Comment (devjgm) [reply ci/kokoro/macos/builds/bazel.sh:88]: Please ignore. I was confused.

**Review by coryan — APPROVED** [review]

**Pull request merged**